### PR TITLE
introduce Zotero

### PR DESCRIPTION
**Pull request by lpranam**
* Target version: master 

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -378,6 +378,10 @@ AC_ARG_ENABLE([experimental],
               AS_HELP_STRING([--enable-experimental],
                              [Enable experimental features and behavior]))
 
+AC_ARG_ENABLE([zotero],
+              AS_HELP_STRING([--enable-zotero],
+                             [Enable control over which uno-commands/feature to disable completely.]))
```
Comment (timar) [inline configure.ac:383]: copy&paste error

File: common/Zotero.hpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+namespace Zotero
+{
+
+const std::string APIURL = "https://api.zotero.org";
+
+class ZoteroConfig
+{
+    static std::string APIKey;
+    static std::string UserId;
+    static bool bla;
```
Comment (timar) [inline common/Zotero.hpp:29]: bla ?

File: common/Zotero.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+#include <Poco/Net/SSLManager.h>
+#include <Poco/Net/KeyConsoleHandler.h>
+#include <Poco/Net/AcceptCertificateHandler.h>
+
+
+namespace Zotero
+{
+std::string ZoteroConfig::APIKey;
+std::string ZoteroConfig::UserId;
+bool ZoteroConfig::bla = false;
```
Comment (timar) [inline common/Zotero.cpp:28]: bla ?

**Review by timar — CHANGES_REQUESTED** [review]
I'm not a big fan of the new configure option `--enable-zotero`, I think we should have it unconditionally. (But perhaps a second review on this would be the best.) It would be a lot of work to add the new configure option to every build script. 
On the other hand, I don't think we can have it mobile apps in the first round, unless it's straightforward. I think the same applies to DeepL and Languagetool integrations , too.

File: common/Zotero.hpp (hunk 2 above)
Comment (lpranam) [reply common/Zotero.hpp:29]: lol sorry about it 😂

File: coolwsd.xml.in [hunk 4]
```diff
@@ -280,6 +280,8 @@
 
     @FEATURE_RESTRICTION_CONFIGURATION@
 
-    @ZOTERO_CONFIGURATION@
+    <zotero dec="Zotero plugin configuration. For more details about Zotero visit https://www.zotero.org/">
+        <enable dec="Enable Zotero plugin." type="bool" default="false">true</enable>
```
Comment (timar) [inline coolwsd.xml.in:284]: change default

**Review by timar — APPROVED** [review]

**Comment by Andreas-Kainz** [comment]
I would prefer a more zotero styled icon (which I can make if you want).

**Pull request merged**